### ISSUE TITLE
Can not suppress LongVariable for single field

### ISSUE THREAD

**Issue by vertexvaar** [phpmd/phpmd#493]
PHPMD Version: 2.6.0

The following code is affected

```PHP
class Authenticator
{
    /**
     * @var AuthenticationService
     * @SuppressWarnings(PHPMD.LongVariable)
     */
    protected $authenticationService = null;
}
```

**What happens**
phpmd reports "Avoid excessively long variable names like $authenticationService. Keep variable name length under 20."

**What should happen**
phpmd should take the SuppressWarnings annotation into account and ignore the length of the annotated field.

**Does it work somewhere else?**
When used in the Class DocBlock PHPMD will ignore all occurrences including fields.

**Comment by ravage84** [comment]
I think we have some recent bug report about suppressions not working.

**Comment by ravage84** [comment]
We had a non-issue here: https://github.com/phpmd/phpmd/issues/403#issuecomment-305944808 plus an actual issue https://github.com/phpmd/phpmd/issues/409

I just fiddled around with your code example:

![phpmd-suppression-bug](https://user-images.githubusercontent.com/625761/27130545-018179cc-5107-11e7-98a5-1292cda25d89.png)

While the suppression in the property doc block does not work, it does when in the class doc block.

**Comment by eeree** [comment]
Actually it's not a bug. SuppressWarnings works per method, class, function or interface. You can apply it either to whole class, method, etc. Here you try to apply it to a single property, which cannot work as it has never been implemented. I'd either close it as unrelated either just mention in Docs, that suppressions works per scope, not per variable/property itself.

**Comment by vertexvaar** [comment]
> cannot work as it has never been implemented
That's a pity, which makes this not a bug but definitely a missing feature.

**Comment by ravage84** [comment]
Actually this seems to be a simple case of documentation issue
https://phpmd.org/documentation/suppress-warnings.html

We are simply not clear enough how it works and what it provides (and what not).

**Comment by MacSim75** [comment]
@eeree It's not a bug but it should be a feature !!
I also met that case and I didn't wanted to suppress the warnings on the whole class just for 1 property which was inherited from another class that is not mine (Drupal 8 core)

**Comment by kylekatarnls** [comment]
The same happens for ShortVariable too:
https://github.com/phpmd/phpmd/issues/337#issuecomment-208363809

**Comment by ravage84** [comment]
@MacSim75 agreed.

@kylekatarnls yes, I just verified that, too. I think we should allow to suppress on property level.

**Comment by TimothyJCowen** [comment]
Any update on this? I would love to be able to suppress a warning for an individual property/parameter.

**Comment by dahse89** [comment]
Any update?

**Comment by tvbeek** [comment]
@dahse89 Feel free to create a PR with a fix. It isn't done until someone writes a fix for this. 😃

**Comment by bkdotcom** [comment]
Any update on this, I would like to be able to suppress warnings on a closure (being passed as a param to a method)

            /**
             * @SuppressWarnings(PHPMD.UnusedFormalParameter)
             */
            static function ($reason, $index, PromiseInterface $aggregate) {
                $aggregate->reject($reason);
            }

**Comment by AJenbo** [comment]
no one has volunteered to work on it yet

**Comment by Muetze42** [comment]
Wow. 8 years later. No fix? :O
